python/tests/spark/sql/codegen/test_mlflow_registry.py

```diff
@@ -0,0 +1,82 @@
+#  Copyright (c) 2021 Rikai Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import mlflow
+import pytest
+from pyspark.sql import Row, SparkSession
+from utils import check_ml_predict
+
+from rikai.spark.sql.codegen.mlflow_registry import (
+    codegen_from_runid,
+    MlflowModelSpec,
+)
+from rikai.spark.sql.schema import parse_schema
+
+
+@pytest.fixture(scope="module")
+def mlflow_client(tmp_path_factory, resnet_model_uri):
+    tracking_uri = (
+        "file://" + str(tmp_path_factory.mktemp("mlflow_tracking")) + "/mlruns"
+    )
+    mlflow.set_tracking_uri(tracking_uri)
+    # simpliest
+    with mlflow.start_run():
+        mlflow.log_param("optimizer", "Adam")
+        mlflow.set_tags(
+            {
+                "rikai.pytest.testcase": "simple",
+                "rikai.spec.version": "1.0",
+                "rikai.model.flavor": "pytorch",
+                "rikai.output.schema": (
+                    "struct<boxes:array<array<float>>, "
+                    "scores:array<float>, labels:array<int>>"
+                ),
+                "rikai.transforms.pre": (
+                    "rikai.contrib.torch.transforms."
+                    "fasterrcnn_resnet50_fpn.pre_processing"
+                ),
+                "rikai.transforms.post": (
+                    "rikai.contrib.torch.transforms."
+                    "fasterrcnn_resnet50_fpn."
+                    "post_processing"
+                ),
+                "rikai.model.artifact.uri": resnet_model_uri,
```

Comment:
we should support `mlflow.<flavor>.load_func/load_model` from udf?

This file based URI might be just an accident support in mlflow

Reply:
You mean like "ML_PREDICT(mlflow.pytorch.<model_name>[.latest | .<version>], ...)"? That's a good idea actually.

Reply:
i meant `https://www.mlflow.org/docs/latest/model-registry.html#fetching-an-mlflow-model-from-the-model-registry` 

The case you mentioned can do too.

Currently `ML_PREDICT` does support in flight model loading like

```
ML_PREDICT("mlflow://models/model_name", ...) 
```

and it will load a temporary model, which is not registered into ModelCatalog

Reply:
oh i see yeah i added that already.